netlify/remix-compute
# Adds Hydrogen 2.0 (via Remix) Adapter

## Description

This PR adds three new workspaces to this project:

1. `@netlify/hydrogen-edge-adapter`
2. `@netlify/hydrogen-netlify-plugin` Netlify build plugin to transform Hydrogen build output.
3. `hydrogen-demo-site` The Hydrogen Remix demo store so we can run E2E tests

This allows running Hydrogen with both the Netlify CLI and the existing Remix/Hydrogen dev server (via `npm run dev`). **Important note** that this includes one change to the remix edge adapter to add a defensive check for the Deno global (which unlocks using the native dev server)—see `packages/remix-edge-adapter/src/globals.ts`.

## QA

All required checks should be green.

1. Ensure the [latest version of the Netlify CLI is installed](https://docs.netlify.com/cli/get-started/).
1. If you want to serve the Remix Hydrogen demo site locally run

    ```bash
    ntl serve --filter demo-store
    ```

## Related Tickets & Documents

* https://github.com/netlify/pod-ecosystem-frameworks/issues/385
* https://github.com/netlify/docs/issues/3181

### Follow up tasks:

1. Archive https://github.com/netlify/hydrogen-platform (the Vite plugin for Hydrogen v1) (or at the very least update the README to make it clear it only works with Hydrogen v1 and link to the v2 stuff)
1. Update https://github.com/netlify/hydrogen-netlify-starter with new demo site (*or* retire and add as a workspace to this repo)
1. Update Netlify docs (see https://github.com/netlify/docs/issues/3181) https://docs.netlify.com/integrations/frameworks/hydrogen/
1. ~File an issue upstream regarding the hardcoded `remix.config.js` values for Hydrogen projects, which are the only reason the build plugin is needed. If we can get rid of these upstream, we can remove the Netlify build plugin. See https://github.com/Shopify/hydrogen/blob/180d7172fe1d16b3f1c34852f8adcdaf32d573ec/packages/cli/src/lib/config.ts#L113~ @nickytonline started a discussion and a PR is up to remove this constraint. See https://github.com/Shopify/hydrogen/pull/1137
